src/pytorch_lightning/utilities/apply_func.py

```diff
@@ -39,6 +40,10 @@
 
 
 _CPU_DEVICES = ("cpu", torch.device("cpu"))
+if _MPS_AVAILABLE:
+    _MPS_DEVICES = ("mps", torch.device("mps"))
```

Comment:
```suggestion
    _MPS_DEVICES = ("mps", torch.device("mps"), torch.device("mps:0"))
```

This is what pytorch creates most of the time and unfortunately, they are not the same

Reply:
I prefer my other suggestion below :) If we don't use that one, we have to do this though :)